### ISSUE TITLE
SPFX: Configure web service urls per environment.

### ISSUE THREAD

**Issue by spoman007** [SharePoint/sp-dev-docs#5534]
If a webpart is calling an external web service that has different urls per environment. Is there any standarard way to build the package for 
1. Dev vs
2. QA

So when deployed to Dev the webpart will call dev endpoint and so on.
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 819a6029-5ccb-0a7b-8c59-a1ebeffca4cb
* Version Independent ID: 61b64d96-30a0-93d8-a0fd-6523e2dd5418
* Content: [SharePoint Framework toolchain](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/toolchain/sharepoint-framework-toolchain)
* Content Source: [docs/spfx/toolchain/sharepoint-framework-toolchain.md](https://github.com/SharePoint/sp-dev-docs/blob/master/docs/spfx/toolchain/sharepoint-framework-toolchain.md)
* Product: **sharepoint**
* Technology: **sharepoint-framework**
* GitHub Login: @spdevdocs
* Microsoft Alias: **spdevdocs**

**Comment by andrewconnell** [comment]
No... there are lots of ways to deal with this. It depends on what you're comfortable with. Personally, I'd use a tenant scoped property.

**Comment by spoman007** [comment]
Thank you @andrewconnell 
I went ahead with creating .env files.
.env.production, .env.development ect.
And while building the package I used cross-env to pass env value

`"bundle:production": "cross-env NODE_ENV=PRODUCTION gulp bundle --ship",`
`"bundle:development": "cross-env NODE_ENV=DEVELOPMENT gulp bundle --ship"`

And finally in gulp file I did something like this
`if (pluginDefine) {
      pluginDefine.definitions = {
        ...pluginDefine.definitions,
        ...currentEnvValues,
      };
    } else {
      cfg.plugins.push(new webpack.DefinePlugin(currentEnvValues));
    }`

Works really fine for me.
